iconic/UserProfile/views.py

```diff
@@ -187,3 +188,61 @@ def get(self, request):
         cities = Cities.objects.all()
         serializer = CitiesSerializer(cities, many=True)
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+# A view for getting all users posts
+class AllUserPostsGetView(APIView):
+    permission_classes = (permissions.AllowAny,)
+
+    def get(self, request):
+        user_id = request.data['user_id']
+        current_posts = Post.objects.filter(user_id=user_id)
+        data = CustomPostSerializer(current_posts).data
+        new_data = []
+        for dic in data:
+            dic["likes"] = len(Likes.objects.filter(post_id=dic["id"]))
+            new_data.append(dic)
+        return Response(new_data, status=status.HTTP_200_OK)
```

Comment:
There is may be more than one objects of currents posts, because of it, you should to write CustomPostSerializer(current_posts, many=True) instead of CustomPostSerializer(current_posts)